jerry-core/ecma/builtin-objects/ecma-builtin-regexp.c

```diff
@@ -78,8 +78,20 @@ ecma_builtin_regexp_dispatch_construct (const ecma_value_t *arguments_list_p, /*
     }
   }
 
-  if (ecma_is_value_object (pattern_value)
-      && ecma_object_class_is (ecma_get_object_from_value (pattern_value), LIT_MAGIC_STRING_REGEXP_UL))
+#if ENABLED (JERRY_ES2015)
+  ecma_value_t regexp_value = ecma_op_is_regexp (pattern_value);
```

Comment:
If we add this, then we should also implement handling RegExp-like objects as arguments, which means creating a builtin RegExp instance from the properties of the argument object, as described in 21.2.3.1/6.